pkg/proxy/types.go

```diff
@@ -42,3 +43,33 @@ type ServicePortName struct {
 func (spn ServicePortName) String() string {
 	return fmt.Sprintf("%s:%s", spn.NamespacedName.String(), spn.Port)
 }
+
+// ServicePort is an interface which abstracts a service information.
+type ServicePort interface {
+	// String returns service string.
+	String() string
+	// ClusterIP returns service cluster IP.
+	ClusterIP() string
+	// Protocol returns service protocol.
+	Protocol() api.Protocol
+	// HealthCheckNodePort returns service health check node port if present.  If return 0, it means not present.
+	HealthCheckNodePort() int
+}
+
+// Endpoints is an interface which abstracts an endpoints info.
+type Endpoints interface {
+	// String returns endpoints string.
+	String() string
+	// IsLocal returns if endpoints is local.
+	IsLocal() bool
+	// IPPart returns IP part of endpoints.
+	IPPart() string
+	// Equal checks if two endpoints are equal.
+	Equal(Endpoints) bool
+}
+
+// EndpointServicePair is an Endpoint and ServicePortName pair.
```

Comment:
This comment is a little redundant. We obviously know EndpointServicePair contains an Endpoint and ServicePortName just by looking at it.

Reply:
Done.